apps/aecore/src/aec_sync.erl

```diff
@@ -294,7 +406,10 @@ process_job([{_T, Job}]) ->
         {forward, #{tx := Tx}, Uri} ->
             do_forward_tx(Tx, Uri);
         {start_sync, Uri, RemoteHash} ->
-            do_start_sync(Uri, RemoteHash);
+            case sync_in_progress(Uri) of
```

Comment:
I notice this as first improvement: it prevents spending effort on finding agreed height if sync for this URI already in progress (same check is done in do_start_sync function - see new_header call - after height agreed).

No change required for this in this PR, though note at https://github.com/aeternity/epoch/pull/829#discussion_r172653557 still applies.

Reply:
Yep, there are one or two calls at the borders that are duplicated, but that's because the chain may have just forked at the top.